receiver/hostmetricsreceiver/README.md

```diff
@@ -83,15 +83,59 @@ network:
 ```
 
 ### Process
+Process can be configured to have multiple filters.  Each filter operates
+on an and basis.  If a process matches any one filter then it will be 
+included, even if another filter explicitly excludes it.
+```yaml
+process:
+  mute_process_name_error: <true|false>
+  filters:
+    -
+      <include_executable_name|exclude_executable_name>:
```

Comment:
This is a broader question and I see we already follow this convention here (e.g. filesystem has `include_devices|exclude_devices` and so forth) but I think that having such "flat" list of includes/excludes is not consistent with other processors. Perhaps that's fine but I just wanted to raise this cc @dmitryax 

For reference, one alternative is e.g. config with this structure (roughly based on [attributesprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/attributesprocessor#includeexclude-filtering)):
```yaml
filters:
  - include:
      executable_names:
        match_type: <strict|regexp> 
        values: [<executable_name>, ... ] 
      executable_paths:
        match_type: <strict|regexp>
        values: [<executable_path>, ... ]
    exclude:
      ...
  - include: ...
```

Or even:
```yaml
filters:
  - include:
      match_type: <strict|regexp>
      executable_names: [<executable_name>, ... ] 
      executable_paths: [<executable_path>, ... ]
    exclude:
      ...
  - include: ...
```

Reply:
I used filesystem as my template since we should have consistency across the hostmetricsreceiver.  I don't have a preference either way so I'll leave that up to @dmitryax.  If we do decide to change this, I think it might be best to do it in a different PR so that this one doesn't get too cluttered.  I agree though, we should follow the same standards for all configurable items.

Reply:
I would like to have the consistency across `hostmetric` scrapers in terms of configuration, but this scraper provides a way to setup several filters as a list which is not consistent.

@davidmirza408 can you please let us know what's the use case you need to cover (maybe a desired configuration snippet) in order to have the list of filters instead one flat filter with `include_...` and `exclude_...` settings as other scrapers have?

If we can't make it flat the same way as done in other scrapers, what your desired config will look like using this PR's version and using what's @pmm-sumo suggested?

```yaml
filters:
  - include:
      match_type: <strict|regexp>
      executable_names: [<executable_name>, ... ] 
      executable_paths: [<executable_path>, ... ]
    exclude:
      ...
  - include: ...
```

Reply:
@dmitryax This use case brings in some of the other tickets regarding filtering options that I have created.  Assume you want to monitor the OtelCollector process, all processes running as root, your service process, and all child processes that your service spins up after running for 60s (to avoid reporting on short lived processes).  Your config would look something like this:

receivers:
  hostmetrics:
    scrapers:
      process:
        filters:
          -
            include_executable_name:
              executable_name: [appd-otel-collector]
              match_type: strict
          -
            include_executable_name:
              executable_name: [my-child-process]
              match_type: strict
            min_uptime: 60s
          -
            include_owner: root

The minuptime and include_owner options has already been developed but I am waiting for approval of the issue ticket I created before submitting a pr for that.

Reply:
> On the backend, reading and storing process metrics can be an expensive job in terms of processing and storing the metrics. My goal with these changes is to give the user as many tools as possible to allow the user to focus the process metrics on exactly the data they need.

BTW, I see value in limiting the metrics on the producer of the data but then it's already possible to filter them using [filter processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/filterprocessor) (and probably soon via transformprocessor)

Reply:
I think there are some advantages to filtering the data in the receiver itself.  It makes the config easier to read since it is clear what is being filtered (i.e. process metrics).  In addition to that we do not run the risk of unintentionally filtering out metrics produced by another receiver that has matching attribute names. 

The size of the metrics being collected by host metrics receiver is very large, depending on what is running on the host.  If we filter it out before it's even created we can save cycles and memory.

Reply:
@davidmirza408 can you please wrap your config with [markdown code block ](https://www.markdownguide.org/extended-syntax/#fenced-code-blocks). Otherwise it's hard to read.

The following configuration interface will be consistent with what we have in other scrapers:

```yaml
receivers:
  hostmetrics:
    scrapers:
      process:
        include_executable_names: 
          - executable_names: [appd-otel-collector, my-child-process]
            match_type: strict
```

Please let us know why it doesn't work for you

Reply:
@davidmirza408 can you please reply to this comment ^?

Reply:
@dmitryax I'm not sure I understand what you are referring too.  I have a markdown code block here https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/8977/files/83c044848c3ebcec9e48b31061227fcc4bb10952#diff-309489c219e376c04b20768f5fd24ffdcf2ed6359bfd13ae0f42d50014452792R89

Is this what you are referring to?

Reply:
@davidmirza408 can you please let us know why the following configuration interface cannot be used?

```yaml
receivers:
  hostmetrics:
    scrapers:
      process:
        include_executable_names: 
          - executable_names: [appd-otel-collector, my-child-process]
            match_type: strict
```

This interface is consistent with all the other scrapers.

Reply:
@dmitryax the above configuration format does not allow us to filter based on multiple attributes at the same time.  Using the example above, if you wanted to filter for "my-child-process" started by user "root"  with a minimum uptime of 10 seconds, that would not be possible.  Allowing the create filters that combine multiple include/exclude statements allows for more flexibility.